pkg/kubelet/volumemanager/cache/desired_state_of_world.go

```diff
@@ -259,7 +274,62 @@ func (dsw *desiredStateOfWorld) AddPodToVolume(
 		volumeName = util.GetUniqueVolumeNameFromSpecWithPod(podName, volumePlugin, volumeSpec)
 	}
 
-	if _, volumeExists := dsw.volumesToMount[volumeName]; !volumeExists {
+	var seLinuxFileLabel string
+	// Volume plugin supports SELinux context mount for all its volumes.
+	var pluginSupportsSELinuxContextMount bool
+	// The volume is ReadWriteOncePod. We don't support other volume types in SELinuxMountReadWriteOncePod feature.
+	// Don't use mount option to apply the SELinux context, still, track the context and report metrics of things
+	// that would break if the feature was for all volume access modes.
+	var isRWOP bool
+
+	if feature.DefaultFeatureGate.Enabled(features.SELinuxMountReadWriteOncePod) {
+		pluginSupportsSELinuxContextMount = dsw.getSELinuxMountSupport(volumeSpec)
+		isRWOP = util.IsRWOP(volumeSpec)
+		if pluginSupportsSELinuxContextMount {
+			// Ensure that a volume that can be mounted with "-o context=XYZ" is
+			// used only by containers with the same SELinux contexts.
+			for _, containerContext := range seLinuxContainerContexts {
+				newLabel, err := util.SELinuxOptionsToFileLabel(containerContext)
+				if err != nil {
+					fullErr := fmt.Errorf("failed to construct SELinux label from context %q: %s", containerContext, err)
+					if isRWOP {
+						// Cannot mount with -o context if the context can't be composed.
+						seLinuxContainerContextErrors.Add(1.0)
```

Comment:
Does it make sense for these metrics to count volume type?

Reply:
Yes, I will add plugin name + the "highest" access mode (a PV with rwx + rwo would get a metric for rwx)

Reply:
Added a TODO item to desired_state_of_wold_selinux_metrics.go